clientv3/integration/maintenance_test.go

```diff
@@ -84,3 +94,61 @@ func TestMaintenanceMoveLeader(t *testing.T) {
 		t.Fatalf("new leader expected %d, got %d", target, lead)
 	}
 }
+
+func TestMaintenanceSnapshotError(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
+	defer clus.Terminate(t)
+
+	clus.Members[0].Stop(t)
```

Comment:
I don't think it is needed to write data cluster. this tests should care about about context canceled and deadline exceeded.

Reply:
We need test canceling the inflight snapshot stream. Without data, it's hard to simulate concurrent context cancel while snapshot is in progress. This manual mvcc writes give us about >1 second window where we can inject concurrent context cancel.